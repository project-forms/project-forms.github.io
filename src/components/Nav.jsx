import {
  ActionList,
  ActionMenu,
  Avatar,
  Header,
  IconButton,
  StyledOcticon,
} from "@primer/react";
import { TableIcon } from "@primer/octicons-react";

export default function Nav({ avatarUrl, onLogout }) {
  return (
    <Header>
      <Header.Item full>
        <Header.Link href="https://github.com/project-forms">
          <StyledOcticon icon={TableIcon} size={32} sx={{ mr: 2 }} />
          <span>Project Forms</span>
        </Header.Link>
      </Header.Item>
      {avatarUrl && (
        <Header.Item>
          <ActionMenu>
            <ActionMenu.Anchor>
              <IconButton
                sx={{ p: 0 }}
                icon={() => <Avatar src={avatarUrl} size={32} />}
                variant="invisible"
              />
            </ActionMenu.Anchor>

            <ActionMenu.Overlay>
              <ActionList onClick={onLogout}>
                <ActionList.Item>Sign Out</ActionList.Item>
              </ActionList>
            </ActionMenu.Overlay>
          </ActionMenu>
        </Header.Item>
      )}
    </Header>
  );
}
