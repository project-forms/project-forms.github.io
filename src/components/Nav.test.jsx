import { describe, expect, test, beforeAll, vi } from "vitest";
import { render } from "@testing-library/react";
import Nav from "./Nav.jsx";

describe("<Nav>", () => {
  // This prevents having the error of ReferenceError: ResizeObserver is not defined
  // https://github.com/jsdom/jsdom/issues/3368#issuecomment-1147970817
  beforeAll(() => {
    global.ResizeObserver = class ResizeObserver {
      observe() {
        // do nothing
      }
      unobserve() {
        // do nothing
      }
      disconnect() {
        // do nothing
      }
    };
  });

  test("renders expected HTML when 'avatarURL' is not provided", async () => {
    const { asFragment } = render(<Nav />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders expected HTML when 'avatarURL' is provided", async () => {
    const { asFragment } = render(
      <Nav avatarUrl={"random-avatar.png"} onLogout={vi.fn()} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
