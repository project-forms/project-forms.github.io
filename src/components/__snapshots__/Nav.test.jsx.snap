// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Nav> > renders expected HTML 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="Header-sc-1wu4agm-0 fCKyxE"
      >
        <div
          class="Header__HeaderItem-sc-1wu4agm-1 hFzBrP"
        >
          <a
            class="Header__HeaderLink-sc-1wu4agm-2 jjjuuj"
            href="https://github.com/project-forms"
          >
            <svg
              aria-hidden="true"
              class="Octicon-sc-9kayk9-0 eoNnjt"
              fill="currentColor"
              focusable="false"
              height="32"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
              viewBox="0 0 24 24"
              width="32"
            >
              <path
                d="M2 3.75C2 2.784 2.784 2 3.75 2h16.5c.966 0 1.75.784 1.75 1.75v16.5A1.75 1.75 0 0 1 20.25 22H3.75A1.75 1.75 0 0 1 2 20.25ZM9 9v11.5h11.25a.25.25 0 0 0 .25-.25V9Zm11.5-1.5V3.75a.25.25 0 0 0-.25-.25H9v4ZM3.5 9v11.25c0 .138.112.25.25.25H7.5V9Zm4-1.5v-4H3.75a.25.25 0 0 0-.25.25V7.5Z"
              />
            </svg>
            <span>
              Project Forms
            </span>
          </a>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="Header-sc-1wu4agm-0 fCKyxE"
    >
      <div
        class="Header__HeaderItem-sc-1wu4agm-1 hFzBrP"
      >
        <a
          class="Header__HeaderLink-sc-1wu4agm-2 jjjuuj"
          href="https://github.com/project-forms"
        >
          <svg
            aria-hidden="true"
            class="Octicon-sc-9kayk9-0 eoNnjt"
            fill="currentColor"
            focusable="false"
            height="32"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 24 24"
            width="32"
          >
            <path
              d="M2 3.75C2 2.784 2.784 2 3.75 2h16.5c.966 0 1.75.784 1.75 1.75v16.5A1.75 1.75 0 0 1 20.25 22H3.75A1.75 1.75 0 0 1 2 20.25ZM9 9v11.5h11.25a.25.25 0 0 0 .25-.25V9Zm11.5-1.5V3.75a.25.25 0 0 0-.25-.25H9v4ZM3.5 9v11.25c0 .138.112.25.25.25H7.5V9Zm4-1.5v-4H3.75a.25.25 0 0 0-.25.25V7.5Z"
            />
          </svg>
          <span>
            Project Forms
          </span>
        </a>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
